tests/FunctionalTestCase.php

```diff
@@ -252,33 +260,65 @@ public static function getModuleInfo(string $row): ?string
     }
 
     /**
-     * Creates the test collection with the specified options.
+     * Creates the test collection with the specified options  and ensures it is dropped again during tearDown().
      *
      * If the "writeConcern" option is not specified but is supported by the
      * server, a majority write concern will be used. This is helpful for tests
      * using transactions or secondary reads.
      */
-    protected function createCollection(array $options = []): void
+    protected function createCollection(?string $databaseName = null, ?string $collectionName = null, array $options = []): void
     {
         $options += ['writeConcern' => new WriteConcern(WriteConcern::MAJORITY)];
 
-        $operation = new CreateCollection($this->getDatabaseName(), $this->getCollectionName(), $options);
+        if (null === $databaseName) {
+            $databaseName = $this->getDatabaseName();
+        }
+
+        if (null === $collectionName) {
+            $collectionName = $this->getCollectionName();
+        }
+
+        $operation = new CreateCollection($databaseName, $collectionName, $options);
         $operation->execute($this->getPrimaryServer());
+        $this->collectionsToCleanup[$databaseName][$collectionName] = true;
     }
 
     /**
-     * Drops the test collection with the specified options.
-     *
-     * If the "writeConcern" option is not specified but is supported by the
-     * server, a majority write concern will be used. This is helpful for tests
-     * using transactions or secondary reads.
+     * Drops the test collection and ensures it is dropped again during tearDown().
      */
-    protected function dropCollection(array $options = []): void
+    protected function dropCollection(?string $databaseName = null, ?string $collectionName = null): void
     {
-        $options += ['writeConcern' => new WriteConcern(WriteConcern::MAJORITY)];
+        $options = ['writeConcern' => new WriteConcern(WriteConcern::MAJORITY)];
 
-        $operation = new DropCollection($this->getDatabaseName(), $this->getCollectionName(), $options);
+        if (null === $databaseName) {
+            $databaseName = $this->getDatabaseName();
+        }
+
+        if (null === $collectionName) {
+            $collectionName = $this->getCollectionName();
+        }
+
+        $operation = new DropCollection($databaseName, $collectionName, $options);
         $operation->execute($this->getPrimaryServer());
+        $this->collectionsToCleanup[$databaseName][$collectionName] = true;
+    }
+
+    protected function cleanupCollections(): void
+    {
+        foreach ($this->collectionsToCleanup as $databaseName => $collectionNames) {
+            // The default database is never dropped.
```

Comment:
I have noticed that when I delete all the collections in a database, the database is no longer listed. Could we delete the default database directly?

Reply:
I think this comment predates our [Slack conversation](https://mongodb.slack.com/archives/C026UA4GF0B/p1684745839221829). Feel free to resolve if so.